vnext/Shared/IHttpResource.h

```diff
@@ -3,32 +3,51 @@
 
 #pragma once
 
-#include <Folly/dynamic.h>
+// Standard Libryary
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
 
 namespace Microsoft::React {
 
 struct IHttpResource {
-  typedef std::map<std::string, std::string> Headers;
+  typedef std::unordered_map<std::string, std::string> Headers;
 
-  static std::unique_ptr<IHttpResource> Make() noexcept;
+  struct BodyData {
+    enum class Type : size_t { Empty, String, Base64, Uri, Form } Type = Type::Empty;
+    std::string Data;
+  };
+
+  struct Response {
+    int64_t StatusCode;
+    Headers Headers;
+    std::string Url;
+  };
+
+  static std::shared_ptr<IHttpResource> Make() noexcept;
 
   virtual ~IHttpResource() noexcept {}
 
   virtual void SendRequest(
       const std::string &method,
       const std::string &url,
-      const Headers &headers,
-      folly::dynamic bodyData, // ISS:2365799 - Make non-folly.
+      const Headers &&headers,
+      BodyData &&bodyData,
```

Comment:
@acoates-ms I followed the same solution you implemented in Office to remove the interface dependency on Folly.

This aims to be a fully in-DLL implementation, thus, it's not intended to be consumed by any external component in production.

Would it make sense to use folly types again to reduce the translation complexity and copying?

Reply:
These types will help when we move away from CxxModules, which I expect we will do for this once we are sure we no longer need the CxxModule for Office.

Reply:
Understood. I'll keep as is.

One unaddressed item is supporting FORM data.
We may need to manually convert/copy a deep data structure.